src/documents/PromisesAPI.md

```diff
@@ -0,0 +1,236 @@
+
+## Core Promises API
+
+### fluid.promise()
+
+* Returns: `{Promise}`
+
+Construct a fresh promise. This is the only point at which fresh promises are constructed within the core API.
+
+### promise.then(onResolve, onReject)
+
+Adds handlers to either or both of the `resolve` or `reject` actions of the promise. Note that if the promise has already been
+rejected or resolved, the appropriate handler will be notified immediately on this registration.
+
+* `onResolve: {Function ({Any}) →  None}` A callback to receive the successfully resolved value of the promise
+* `onReject: {Function ({Error}) →  None}` A callback to receive a rejection of the promise, in the case its resolution fails.
+
+### promise.resolve(value)
+
+Resolves the promise successfully, yielding the value `value` to any listeners which were previously registered as the first
+argument of `promise.then`. Any further attempts to call either `resolve` or `reject` will signal an assertion failure.
+
+* `value: {Any}` The value to be supplied as the resolution value of the promise
+
+### promise.reject(error)
+
+Rejects the promise, yielding the error `error` to any listeners which were previously registered as the second 
+argument of `promise.then`. Any further attempts to call either `resolve` or `reject` will signal an assertion failure.
+
+* `error: {Object|Error}` The value to be supplied as the rejection reason for the promise. It is recommended that this be an
+object with a boolean entry `isError: true` and a String `message` summarising the reason for the rejection. 
+
+### promise.disposition
+
+The current disposition of the promise may be inspected at any time. This is a `String` value which encodes which, if any, of
+`resolve` or `reject` have been received by the promise. At the fresh construction of the promise, the member `disposition` holds
+the value `undefined`. If the promise has received `resolve`, `disposition` will hold the string `"resolve"`, or if the
+promise has received `reject`, `disposition` will hold the string `"reject"`.
+
+### Commentary on core API
+
+This is 100% of the core implementation of Infusion promises. It puts us in a position to answer, in terms of familiar constructions
+such as [events](InfusionEventSystem.md), the question "What actually is a promise?" (although not in itself the more important question
+"What can a promise be used for?")
+
+A promise is
+
+* A linked pair of event firers, named `resolve` and `reject`
+* At most one of these two events can be fired, at most one time in total
+* Any listeners registered to either of the events after the point of firing will be able to recover the (unique) fired value at the point of registration
+
+Note that this description does not adequately account for the features of specification-conformant promises more widely used in the industry - since these
+implement chaining and asynchronous behaviours which we do not implement.
+
+### Commentary on requirements
```

Comment:
For some reason I can't see this comment anywhere in the github interface. Here is the argument:

Given that the API structure, both in name and signature, of our promises API, is identical to that of other 
libraries widely available, but at the same time the semantic and behaviour is quite different, it would be 
actively misleading to confine our mission in this "reference material" to simply describing these minutiae 
of the API form. A user who simply takes the form and description of our API and tries to use it, with the 
expectations that they have brought from some other Promises API, without understanding the differences in 
philosophy and intention lying behind our API, would run into serious problems.

On 02/11/2015 19:52, Anastasia Cheetham wrote:
> In src/documents/PromisesAPI.md <https://github.com/fluid-project/infusion-docs/pull/77#discussion_r43673024>:
>
>> +### Commentary on core API
>> +
>> +This is 100% of the core implementation of Infusion promises. It puts us in a position to answer, in terms of familiar constructions
>> +such as [events](InfusionEventSystem.md), the question "What actually is a promise?" (although not in itself the more important question
>> +"What can a promise be used for?")
>> +
>> +A promise is
>> +
>> +* A linked pair of event firers, named `resolve` and `reject`
>> +* At most one of these two events can be fired, at most one time in total
>> +* Any listeners registered to either of the events after the point of firing will be able to recover the (unique) fired value at the point of registration
>> +
>> +Note that this description does not adequately account for the features of specification-conformant promises more widely used in the industry - since these
>> +implement chaining and asynchronous behaviours which we do not implement.
>> +
>> +### Commentary on requirements
>
> @amb26 <https://github.com/amb26>, I guess I'm thinking that it's not appropriate for /API/ documentation.
> Maybe it's just my own mental model; to me, "API documentation" is /reference/ material. You turn to it to
> look up the specifics of the API, not to develop a deeper understanding of the design of the functionality.
> I do believe it's valuable information, but its tone, its purpose... that seems inconsistent with the
> otherwise brief, direct assistance offered by this API page.
>
> But I could be convinced otherwise by good arguments :-)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fluid-project/infusion-docs/pull/77/files#r43673024>.
>